packages/ra-ui-materialui/src/button/Button.tsx

```diff
@@ -38,7 +32,9 @@ export type LocationDescriptor = Partial<Path> & {
  * </Button>
  *
  */
-export const Button = (props: ButtonProps) => {
+export const Button = <RecordType extends RaRecord = RaRecord>(
```

Comment:
should default to any

Reply:
Actually, I failt to see why there is a `record` prop. This should be handled by the specialized buttons only imo